### ISSUE TITLE
`np.NINF` was removed in the NumPy 2.0 release. Use `-np.inf` instead

### ISSUE THREAD

**Issue by tam-nguyenminh** [rasbt/mlxtend#1100]
Hello,

I got this error while trying to use SequentialFeatureSelector. 
Could you please help take a look in the underlying codes?
Thanks a lot

`sfs = SFS(LogisticRegression(), 
          k_features=5, #number of features to select
          forward = True, #rolling forward selection
          floating=False, # No floating selection
          scoring='accuracy',
          verbose=5,
          cv=5
          ).fit(X_train,y_train)`

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[214], [line 8](vscode-notebook-cell:?execution_count=214&line=8)
      [1](vscode-notebook-cell:?execution_count=214&line=1) sfs = SFS(LogisticRegression(), 
      [2](vscode-notebook-cell:?execution_count=214&line=2)           k_features=5, #number of features to select
      [3](vscode-notebook-cell:?execution_count=214&line=3)           forward = True, #rolling forward selection
      [4](vscode-notebook-cell:?execution_count=214&line=4)         #   floating=False, # No floating selection
      [5](vscode-notebook-cell:?execution_count=214&line=5)           scoring='accuracy',
      [6](vscode-notebook-cell:?execution_count=214&line=6)           verbose=5,
      [7](vscode-notebook-cell:?execution_count=214&line=7)           cv=5
----> [8](vscode-notebook-cell:?execution_count=214&line=8)           ).fit(X_train,y_train)

File c:\Users\84393\miniconda3\envs\venv_ds\Lib\site-packages\mlxtend\feature_selection\sequential_feature_selector.py:650, in SequentialFeatureSelector.fit(self, X, y, groups, **fit_params)
    [648](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:648) else:
    [649](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:649)     self.fitted = True  # the completion of sequential selection process.
--> [650](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:650)     self.finalize_fit()
    [652](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:652) return self

File c:\Users\84393\miniconda3\envs\venv_ds\Lib\site-packages\mlxtend\feature_selection\sequential_feature_selector.py:655, in SequentialFeatureSelector.finalize_fit(self)
    [654](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:654) def finalize_fit(self):
--> [655](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:655)     max_score = np.NINF
    [656](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:656)     for k in self.subsets_:
    [657](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:657)         if (
    [658](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:658)             k >= self.min_k
    [659](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:659)             and k <= self.max_k
    [660](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:660)             and self.subsets_[k]["avg_score"] > max_score
    [661](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/mlxtend/feature_selection/sequential_feature_selector.py:661)         ):

File c:\Users\84393\miniconda3\envs\venv_ds\Lib\site-packages\numpy\__init__.py:411, in __getattr__(attr)
    [408](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/numpy/__init__.py:408)     raise AttributeError(__former_attrs__[attr])
    [410](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/numpy/__init__.py:410) if attr in __expired_attributes__:
--> [411](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/numpy/__init__.py:411)     raise AttributeError(
    [412](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/numpy/__init__.py:412)         f"`np.{attr}` was removed in the NumPy 2.0 release. "
    [413](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/numpy/__init__.py:413)         f"{__expired_attributes__[attr]}"
    [414](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/numpy/__init__.py:414)     )
    [416](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/numpy/__init__.py:416) if attr == "chararray":
    [417](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/numpy/__init__.py:417)     warnings.warn(
    [418](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/numpy/__init__.py:418)         "`np.chararray` is deprecated and will be removed from "
    [419](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/numpy/__init__.py:419)         "the main namespace in the future. Use an array with a string "
    [420](file:///C:/Users/84393/miniconda3/envs/venv_ds/Lib/site-packages/numpy/__init__.py:420)         "or bytes dtype instead.", DeprecationWarning, stacklevel=2)

AttributeError: `np.NINF` was removed in the NumPy 2.0 release. Use `-np.inf` instead.

**Comment by Ontheroad123** [comment]
modofied  655 "max_score = np.NINF" to "max_score = np.inf"

**Comment by shobhit120801** [comment]
> modofied 655 "max_score = np.NINF" to "max_score = np.inf"

do we have to change it manually ?

**Comment by cwoodside1278** [comment]
Hey @shobhit120801 I ran into the exact same error as you. @Ontheroad123 I did update the source code manually to np.inf or even -np.inf and neither work. I keep getting the same error above even though I have updated the source code

**Comment by Keeva1** [comment]
I don't know anything about codes
Brad

**Comment by d-kleine** [comment]
> Hey @shobhit120801 I ran into the exact same error as you. @Ontheroad123 I did update the source code manually to np.inf or even -np.inf and neither work. I keep getting the same error above even though I have updated the source code

It worked for me, but you need to replace two lines in the source code:
 ```python
 def finalize_fit(self):
        max_score = -np.inf # changed
        ...
        k_score = max_score
        if k_score == -np.inf: # changed
        ...
```

If you anyone of you needs help with this, let me know. I can also push a PR on this.

**Comment by baimamboukar** [comment]
What finally worked for you guys ?

I ran into the exact same issue as @tam-nguyenminh 
```python
sfs_forward = SFS(linear_model,
                  k_features='best',
                  forward=True,
                  floating=False,
                  scoring='neg_mean_squared_error',
                  cv=5)

sfs_forward = sfs_forward.fit(predictor_variables, dependent_variable) # ---> 
#🔴This line has an error (AttributeError: `np.NINF` was removed in the NumPy 2.0 release. Use `-np.inf` instead.Cell Execution Error)
```

I am not sure `what to update` to make it work. @d-kleine

**Comment by d-kleine** [comment]
You need to edit the **source code**, as provided above. 

Clone the repo locally, apply the changes - or checkout my PR at `gh pr checkout 1107` locally. Then install the repo locally, e.g. `. pip install mlxtend` (open a command shell, activate the relevant Python env, nagivate to the working dir where the cloned `mlxtend` folder is located in, e.g. your Desktop, and then execute the command)

**Comment by d-kleine** [comment]
A new version of `mlxtend` has just been released (0.23.2), you can update it via `pip install --upgrade mlxtend`. Once updated, above error message should not pop up anymore.
